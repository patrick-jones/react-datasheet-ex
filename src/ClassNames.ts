export const READ_ONLY_CELL = 'cell read-only';
export const ACTION_CELL = 'rdx-action-cell';

export const HEADER_CELL = 'rdx-header-cell';
export const HEADER_CELL_LEFT = 'rdx-header-cell--left';
export const HEADER_CELL_CENTER = 'rdx-header-cell--center';
export const HEADER_CELL_RIGHT = 'rdx-header-cell--right';

export const HEADER_CELL_CONTAINER = 'rdx-header-cell-container';
export const HEADER_CELL_TITLE = 'rdx-header-cell-title';

export const HEADER_DRAG_SOURCE = 'rdx-header-drag-source';
export const HEADER_DROP_TARGET = 'rdx-header-drop-target';

export const ROW = 'rdx-row';
export const ROW_DROP_TARGET = 'rdx-row-drop-target';
export const ROW_DRAG_SOURCE = 'rdx-row-drag-source';

export const RESIZING_WIDTH = 'rdx-ex-resizing-width';
export const DATA_GRID_RESIZABLE = 'rdx-data-grid-resizable';
export const WIDTH_RESIZE_HANDLE = 'rdx-width-resize-handle';

export const SCROLL_PANEL = 'rds-ex-scroll-panel';
