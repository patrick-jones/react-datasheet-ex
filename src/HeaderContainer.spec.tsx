import React from 'react';
import 'jest';
import {shallow} from 'enzyme';
import toJson from 'enzyme-to-json';
import {HeaderContainer} from './index';
import {HEADER_CELL_CONTAINER} from './ClassNames';


describe('HeaderContainer', () => {

  it('Should render defaults properly', () => {
    const wrapper = shallow(<HeaderContainer>content</HeaderContainer>);
    expect(wrapper.prop('className')).toEqual(HEADER_CELL_CONTAINER);
    expect(wrapper.prop('style')).toBeUndefined();
  });

  it('Should render options properly', () => {
    const className = 'foo-bar';
    const style = {background: 'blue'};
    const wrapper = shallow(
      <HeaderContainer className={className} style={style}>content</HeaderContainer>
    );

    expect(wrapper.prop('className')).toEqual(`${HEADER_CELL_CONTAINER} ${className}`);
    expect(wrapper.prop('style')).toEqual(style);
  });

  it('matches snapshot', () => {
    const className = 'foo-bar';
    const style = {background: 'blue'};
    const wrapper = shallow(
      <HeaderContainer className={className} style={style}>content</HeaderContainer>
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });

});
