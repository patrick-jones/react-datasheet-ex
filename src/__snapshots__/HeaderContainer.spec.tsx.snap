// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderContainer matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <HeaderContainer
    className="foo-bar"
    style={
      Object {
        "background": "blue",
      }
    }
  >
    content
  </HeaderContainer>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "content",
      "className": "rdx-header-cell-container foo-bar",
      "style": Object {
        "background": "blue",
      },
    },
    "ref": null,
    "rendered": "content",
    "type": "span",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "content",
        "className": "rdx-header-cell-container foo-bar",
        "style": Object {
          "background": "blue",
        },
      },
      "ref": null,
      "rendered": "content",
      "type": "span",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
